File: backend/app/models/post.py
from pydantic import BaseModel
from datetime import datetime

class Post(BaseModel):
    id: int
    user: str
    caption: str
    created_at: datetime
    image_url: str | None = None
    video_url: str | None = None
    likes: int = 0
    liked: bool = False
    categories: list[str] = []
    bubble_color: str | None = None


class PostRequest(BaseModel):
    user: str
    caption: str
    image_url: str | None = None
    categories: list[str] = []

File: frontend/lib/models/post.dart
class Post {
  final int id;
  final String user;
  final String caption;
  final DateTime createdAt;
  final String? imageUrl;
  final String? videoUrl;
  final int likes;
  final bool liked;

  Post({
    required this.id,
    required this.user,
    required this.caption,
    required this.createdAt,
    this.imageUrl,
    this.videoUrl,
    this.likes = 0,
    this.liked = false,
  });

  factory Post.fromJson(Map<String, dynamic> json) => Post(
        id: json['id'] as int,
        user: json['user'] as String,
        caption: json['caption'] as String,
        createdAt: DateTime.parse(json['created_at'] as String),
        imageUrl: json['image_url'] as String?,
        videoUrl: json['video_url'] as String?,
        likes: json['likes'] as int? ?? 0,
        liked: json['liked'] as bool? ?? false,
      );
}
